src/odemis/driver/xt_client.py

```diff
@@ -1116,18 +1123,10 @@ def __init__(self, name, role, parent, rng=None, **kwargs):
     def _updatePosition(self, raw_pos=None):
         """
         update the position VA
-        raw_pos (dict str -> float): the position in mm (as received from the SEM)
+        raw_pos (dict str -> float): the position (as received from the SEM). If None is passed the current position is
+            requested from the SEM.
         """
-        if raw_pos is None:
-            position = self.parent.get_stage_position()
-            x, y, z = position["x"], position["y"], position["z"]
-        else:
-            x, y, z = raw_pos["x"], raw_pos["y"], raw_pos["z"]
-
-        pos = {"x": x,
-               "y": y,
-               "z": z,
-               }
+        pos = self.parent.get_stage_position() if not raw_pos else raw_pos
```

Comment:
Does the other way around not make more sense?
> pos = raw_pos if raw_pos else self.parent.get_stage_position()

Reply:
yes, that looks better. Switched!